backend/api/exceptions.py

```diff
@@ -1,33 +1,23 @@
 from rest_framework.views import exception_handler
-from rest_framework.exceptions import AuthenticationFailed, NotAuthenticated, PermissionDenied
-
-def auth_exception_handler(exc, context):
-    """
-    This custom Django REST Framework exception handler adds the
-    X-Reason header for error responses.
-    The header can be used to distinguish "not authenticated" errors
-    ana "permission denied" errors. It is necessary, because currently
-    unauthenticated requests return the 403 Forbidden status code,
-    not 401 Unauthorized. See:
-    https://www.django-rest-framework.org/api-guide/authentication/#sessionauthentication
-
-    Please note that setting the status code to 401 for all "not authenticated" errors
-    would, per https://datatracker.ietf.org/doc/html/rfc7235#section-3.1,
-    require setting the WWW-Authenticate header.
-    """
+from rest_framework.exceptions import ValidationError
 
+def custom_exception_handler(exc, context):
+    #TODO add more comments
     response = exception_handler(exc, context)
```

Comment:
Do not use the default `exception_handler`, you're overriding the response anyways.
Create and return a new instance of `Response`, use the `status_code` field of `ApiError`.